app/views/edit.erb

```diff
@@ -18,57 +18,57 @@
       <div class="col-md-10">
       <span class="label label-default">Birthdate</span>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][month]" value="<%= @robot.birthdate["month"] %>">
+          <input type="text" class="form-control" name="robot[birthdate][month]" value="<%= @robot.birthdate.split("/").first.to_i %>" maxlength="2" required="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][day]" value="<%= @robot.birthdate["day"] %>">
+          <input type="text" class="form-control" name="robot[birthdate][day]" value="<%= @robot.birthdate.split("/")[1].to_i %>" maxlength="2" required="true">
         </div>
         <div class="col-md-2">
-          <input type="text" class="form-control" name="robot[birthdate][year]" value="<%= @robot.birthdate["year"] %>">
+          <input type="text" class="form-control" name="robot[birthdate][year]" value="<%= @robot.birthdate.split("/").last.to_i %>" minlength="4" maxlength="4" required="true">
```

Comment:
the behavior of `@robot.birthdate.split("/")[1]` could live in a `find_year` method. same with the calculations on line 24 and 21